### PR TITLE
[docs][yba][2.20] OIDC with Azure AD

### PR DESCRIPTION
**Pull request by ddhodge**
Add OIDC configuration in YBA for Azure.
DOC-130

@netlify /preview/yugabyte-platform/security/authentication/

### CODE REVIEW

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+"awd": "e12c03b1-7463-8e23-94d2-8d71f17ab99b",
+"iss": "https://login.microsoftonline.com/733dee2h-cb2e-41ab-91f2-29e2af034ffe/v2.0",
+"iat": 1669657223, "nbf": 1669657223, "exp": 1669651124,
+"groups": [
+  "a12d04b1-7463-8e23-94d2-8d71f17ab99b",
+  "c22b03b1-2746-8d34-54b1-6d32f17ba36b",
+],
+"oid": "b12a45b2-7463-5e76-32d3-9d31f15ab77b",
+"sub": "C12a45bGc463_ADLN632d39d31f15ab77b",
+"tid": "a42a45c3-4728-4f98-25d3-6d63f15ab36c",
```
[the remark this answers is not in the document]
Comment (ddhodge) [reply docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:53]: I entered random values

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+```
+
+For more information on configuring group claims and app roles, refer to [Configuring group claims and app roles in tokens](https://learn.microsoft.com/en-us/security/zero-trust/develop/configure-tokens-group-claims-app-roles) in the Azure documentation.
+
+## Set up OIDC with AAD on YugabyteDB Anywhere
+
+To enable OIDC authentication with AAD, you need to do the following:
+
+- Create an app registration in AAD - The AAD IdP configuration includes application registration (registering YugabyteDB Anywhere in the AAD tenant) and configuring AAD to send (redirect) tokens with the required claims to YugabyteDB Anywhere.
+- Configure OIDC in Yugabyte Anywhere - The OIDC configuration uses the application you registered. You can also configure YBA to display the user's JSON Web Token (JWT) on the login screen.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:96]:
```suggestion
- Configure OIDC in YugabyteDB Anywhere - The OIDC configuration uses the application you registered. You can also configure YBA to display the user's JSON Web Token (JWT) on the login screen.
```

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+1. Click **Register**.
+
+    After the application is registered, you can obtain the tenant ID and client ID.
+
+1. Click **Add a certificate or secret**.
+
+1. Select **Client secrets** and click **New client secret**.
+
+1. Enter a description and set the expiry for the secret, then click **Add**.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:121]:
```suggestion
1. Enter a description and set the expiry for the secret, and then click **Add**.
```

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+1. Navigate to **Admin > Advanced > Global Configuration**.
+
+1. Search on OIDC to display the configuration setting and set it to true.
+
+    ![Configuring yb.security.oidc_feature_enhancements flag for OIDC](/images/yp/security/oidc-azure-globalfeature.png)
+
+#### Enable OIDC authentication
+
+To configure YugabyteDB Anywhere for OIDC, you need to be signed on as a superuser. You need your Azure application client ID, client secret, and tenant ID.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:143]:
```suggestion
To configure YugabyteDB Anywhere for OIDC, you need to be signed in as a superuser. You need your Azure application client ID, client secret, and tenant ID.
```

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+    ```sh
+    login.microsoftonline.com/<tenant_id>/v2.0/.well-known/openid-configuration
+    ```
+
+1. Set the scope to `openid email profile`.
+
+1. Set the email attribute to a name for the property to be used in the mapping file, such as `preferred_username`.
+
+1. Select the **Display JWT token on login** option to allow users to access their JWT from the Yugabyte Anywhere sign in page. This allows a user to view and copy their JWT without signing in to YBA. (This option is only available if you enabled the yb.security.oidc_feature_enhancements configuration flag.)
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:161]:
```suggestion
1. Select the **Display JWT token on login** option to allow users to access their JWT from the YugabyteDB Anywhere sign in page. This allows a user to view and copy their JWT without signing in to YBA. (This option is only available if you enabled the `yb.security.oidc_feature_enhancements` configuration flag.)
```

File: docs/content/preview/yugabyte-platform/security/authentication/ldap-authentication-platform.md [hunk 6]
```diff
@@ -39,17 +35,17 @@ To bind to the LDAP server using TLS, include the `ldaptls=1` option in the `ysq
...
 
 After the user is found, to verify that the login is correct, the server disconnects and rebinds to the directory as this user using the password specified by the client.
 
-For more information on the `ysql_hba_conf_csv` flag, refer to [--ysql_hba_conf_csv flag](../../../reference/configuration/yb-tserver/#ysql-hba-conf-csv). For more information on HBA, refer to [Host-based authentication](../../../secure/authentication/host-based-authentication/).
+For more information on the `ysql_hba_conf_csv` flag, refer to [--ysql_hba_conf_csv flag](../../../../reference/configuration/yb-tserver/#ysql-hba-conf-csv). For more information on HBA, refer to [Host-based authentication](../../../../secure/authentication/host-based-authentication/).
 
 ## Example
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/security/authentication/ldap-authentication-platform.md:50]: line 98 - link for Troubleshooting is broken.

**Review by aishwarya24 — APPROVED** [review]
minor comments. LGTM

**Review by rajmaddy89 — APPROVED** [review]
LGTM !!

**Pull request merged**